packages/react-components/react-popover/library/src/components/Popover/usePopover.ts

```diff
@@ -111,20 +111,39 @@ export const usePopover_unstable = (props: PopoverProps): PopoverState => {
   });
 
   const { findFirstFocusable } = useFocusFinders();
+  const activateModal = useActivateModal();
+  const [setActivateModalTimeout] = useTimeout();
 
   React.useEffect(() => {
     if (props.unstable_disableAutoFocus) {
       return;
     }
 
-    if (open && positioningRefs.contentRef.current) {
-      const containerTabIndex = positioningRefs.contentRef.current.getAttribute('tabIndex') ?? undefined;
-      const firstFocusable = isNaN(containerTabIndex)
-        ? findFirstFocusable(positioningRefs.contentRef.current)
-        : positioningRefs.contentRef.current;
+    const contentElement = positioningRefs.contentRef.current;
+
+    if (open && contentElement) {
+      const shouldFocusContainer = !isNaN(contentElement.getAttribute('tabIndex') ?? undefined);
+      const firstFocusable = shouldFocusContainer ? contentElement : findFirstFocusable(contentElement);
+
       firstFocusable?.focus();
+
+      if (shouldFocusContainer) {
+        // Modal activation happens automatically when something inside the modal is focused programmatically.
+        // When the container is focused, we need to activate the modal manually, and we do it on the next tick,
+        // because on this tick the element is just created and Tabster has not instantiated Modalizer yet.
+        setActivateModalTimeout(() => {
```

Comment:
can the timeout and the handling that modalizer is not created yet, but handled by the `useActivateModal` hook?

Reply:
Would like to keep that tabster detail in react-tabster if possible

Reply:
Sure, I can move it there. Probably with optional `deferred` prop for the `useActivateModal` hook (it is not needed in all the cases). Would that be ok?

Reply:
Or given that this is the only use case so far, I can just bake it there and add a prop later if it is ever needed.

Reply:
Done, moved it to the hook and adjusted the reasoning comment.